src/modal_dialog/index.ts

```diff
@@ -0,0 +1,264 @@
+/**
+ * You can customize the initial state of the module from the editor initialization, by passing the following [Configuration Object](https://github.com/artf/grapesjs/blob/master/src/modal_dialog/config/config.js)
+ * ```js
+ * const editor = grapesjs.init({
+ *  modal: {
+ *    // options
+ *  }
+ * })
+ * ```
+ *
+ * Once the editor is instantiated you can use its API. Before using these methods you should get the module from the instance
+ *
+ * ```js
+ * const modal = editor.Modal;
+ * ```
+ *
+ * ## Available Events
+ * * `modal:open` - Modal is opened
+ * * `modal:close` - Modal is closed
+ * * `modal` - Event triggered on any change related to the modal. An object containing all the available data about the triggered event is passed as an argument to the callback.
+ *
+ * ## Methods
+ * * [open](#open)
+ * * [close](#close)
+ * * [isOpen](#isopen)
+ * * [setTitle](#settitle)
+ * * [getTitle](#gettitle)
+ * * [setContent](#setcontent)
+ * * [getContent](#getcontent)
+ * * [onceClose](#onceclose)
+ * * [onceOpen](#onceopen)
+ *
+ * @module Modal
+ */
+
+import { EventHandler } from 'backbone';
+import { debounce, isFunction, isString } from 'underscore';
+import { Module } from '../abstract';
+import EditorModel from '../editor/model/Editor';
+import { createText } from '../utils/dom';
+import defaults from './config/config';
+import ModalM from './model/Modal';
+import ModalView from './view/ModalView';
+
+export default class ModalManager extends Module<typeof defaults> {
+  modal?: ModalView;
+
+  /**
+   * Initialize module. Automatically called with a new instance of the editor
+   * @param {Object} config Configurations
+   * @private
+   */
+  constructor(em: EditorModel) {
+    super(em, 'Modal', defaults);
+
+    this.model = new ModalM(this);
+    this.model.on('change:open', (enable: boolean) => {
+      em.trigger(`modal:${enable ? 'open' : 'close'}`);
+    });
```

Comment:
The first argument of the callback is the model itself and the value is the second one.